docs/components/camera/esp32-camera.md

```diff
@@ -0,0 +1,192 @@
+
+{{< /alert >}}
+
+{{< tabs name="Configure a esp32-camera" >}}
+{{% tab name="JSON Template" %}}
+
+Navigate to the **CONFIGURE** tab of your machine's page in [the Viam app](https://app.viam.com).
+Select **Builder** mode.
+Copy and paste the following JSON into your existing machine configuration in the `"components"` level:
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+    "name": "my-esp32camera",
+    "namespace": "rdk",
+    "type": "camera",
+    "model": "esp32-camera",
+    "attributes": {
+        "pin_d4": <int>,
+        "jpeg_quality": <int>,
+        "frame_size": <int>,
+        "pin_d5": <int>,
+        "pin_d3": <int>,
+        "pin_d6": <int>,
+        "pin_vsync": <int>,
+        "ledc_timer": <int>,
+        "pin_d7": <int>,
+        "pin_sccb_sda": <int>,
+        "pin_href": <int>,
+        "pin_sccb_scl": <int>,
+        "sccb_i2c_port": <int>,
+        "pin_d1": <int>,
+        "pin_d0": <int>,
+        "pin_xclk": <int>,
+        "pin_reset": <int>,
+        "pin_pclk": <int>,
+        "pin_d2": <int>,
+        "xclk_freq_hz": <int>,
+        "ledc_channel": <int>
+    }
+}
+```
+
+Edit and fill in the attributes as applicable.
+
+{{% /tab %}}
+{{% tab name="JSON Example: OV2640" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+  "name": "my-esp32camera",
+  "namespace": "rdk",
+  "type": "camera",
+  "model": "esp32-camera",
+  "attributes": {
+    "pin_pwdn": -1,
+    "pin_reset": -1,
+    "pin_xclk": 21,
+    "pin_sccb_sda": 26,
+    "pin_sccb_scl": 27,
+    "pin_d7": 35,
+    "pin_d6": 34,
+    "pin_d5": 39,
+    "pin_d4": 36,
+    "pin_d3": 19,
+    "pin_d2": 18,
+    "pin_d1": 5,
+    "pin_d0": 4,
+    "pin_vsync": 25,
+    "pin_href": 23,
+    "pin_pclk": 22,
+    "xclk_freq_hz": 20000000,
+    "ledc_timer": 1,
+    "ledc_channel": 1
+  }
+}
+```
+
+{{% /tab %}}
+{{% tab name="JSON Example: OV3660" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+  "name": "my-esp32camera",
+  "namespace": "rdk",
+  "type": "camera",
+  "model": "esp32-camera",
+  "attributes": {
+    "pin_d4": 39,
+    "jpeg_quality": 32,
+    "frame_size": 5,
+    "pin_d5": 18,
+    "pin_d3": 5,
+    "pin_d6": 36,
+    "pin_vsync": 22,
+    "ledc_timer": 1,
+    "pin_d7": 19,
+    "pin_sccb_sda": 25,
+    "pin_href": 26,
+    "pin_sccb_scl": 23,
+    "sccb_i2c_port": -1,
+    "pin_d1": 35,
+    "pin_d0": 32,
+    "pin_xclk": 27,
+    "pin_reset": 15,
+    "pin_pclk": 21,
+    "pin_d2": 34,
+    "xclk_freq_hz": 20000000,
+    "ledc_channel": 1
+  }
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+{{% alert title="Note" color="note" %}}
+While the following attributes marked as **Optional** do have defaults, it is still recommended that you configure them to your datasheet and device's needs (which likely do not align with the defaults) in order to get the camera stream.
```

Comment:
@mattjperez can you confirm this? The example ones worked fine for my esp32-wrover-dev. Are they really likely to not align?

Reply:
until we get more boards (with ribbon ports) and cameras and test, I don't know how likely it is to be honest. It might be standardized or it might not. Either way, users should double check so they don't accidentally break their board and/or camera.

Reply:
maybe change from `which likely do not align with defaults` to `which may not align with the defaults, possibly damaging your board and/or camera`. the damage part is kind of a given in this space, but not sure how much that has been emphasized up to this point

Reply:
done